### DISCUSSION TITLE
How to Get Started with Contributing to Spring AI?

### DISCUSSION THREAD

**Discussion by iamvishnusingh** [spring-projects/spring-ai#1333] [Q&A]
I'm new to contributing and am interested in getting involved with Spring AI. Could someone guide me on how to start? Specifically, I'd like to know:
- Are there any official **documentation** or resources available for new contributors?
- How can **I set up my development environment** to begin contributing?
Any tips or pointers would be greatly appreciated. Thanks in advance for your help!

**Comment by devholic22** [comment]
@iamvishnusingh I’m not sure if you’ll see this since it’s been a month since you posted your question, but I’d like to share my experience.

The simplest way to contribute is by correcting the syntax of comments. Sometimes there are typos as well.
Another way is to use Spring AI yourself and identify any inconveniences you encounter. You can also watch official Spring AI videos or Dan Vega's videos on YouTube to find issues. For example, while comparing the ChatResponse and ImageResponse files, I discovered that an NPE could occur in the ImageResponse.
You can also look at issues submitted by others and ask if you can work on any that you feel you could resolve.
For a more advanced contribution, you might consider adding features, models, or writing test code.
To contribute, identify a problem, create an issue (checking first if it’s already been reported), then fork the repository, create a branch that corresponds to the issue name, do your work, and finally submit a pull request (from your branch to the main branch of Spring AI).

Here are some of my records for your reference:

https://github.com/spring-projects/spring-ai/pull/745
https://github.com/spring-projects/spring-ai/pull/975
https://github.com/spring-projects/spring-ai/pull/1000

**Reply by ogbozoyan** [reply]
Helpful
